mantidproject/mantid
# ReflectometryReductionOne flood correction property

This update adds a new property `DetectorEfficiencyCorrection` - an input workspace that applies flood corrections. ~~This property replaces the `RegionOfDirectBeam` input property and replaces the `Multidetector?` and `Perform Direct Beam Normalization` stages in the workflow diagram for ReflectometryReductionOne.~~ It was later instead decided that this property as well as those stages would remain after all. In addition, the input workspace is now converted to units of wavelength before being split into detectors and monitor workspaces.

**To test:**

~~The old tests involving `RegionOfDirectBeam` have been removed~~. Two new tests was added to to account for the new `DetectorEfficiencyCorrection` property. They each test that:
- The flood correction workspace has the same number of spectra as the input workspace. 
- The flood correction workspace does not have more than one value per spectra.

Simply run the tests to ensure that they pass.

Fixes #17085.
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
